koku/masu/external/accounts/hierarchy/aws/aws_org_unit_crawler.py

```diff
@@ -0,0 +1,147 @@
+#
+"""AWS org unit crawler."""
+# from tenant_schemas.utils import schema_context
+import logging
+
+from django.db import transaction
+from tenant_schemas.utils import schema_context
+
+from masu.external.accounts.hierarchy.account_crawler import AccountCrawler
+from masu.external.date_accessor import DateAccessor
+from masu.util.aws import common as utils
+from reporting.provider.aws.models import AWSOrganizationalUnit
+
+LOG = logging.getLogger(__name__)
+
+
+class AWSOrgUnitCrawler(AccountCrawler):
+    """AWS org unit crawler."""
+
+    def __init__(self, account):
+        """
+        Object to crawl the org unit structure for accounts to org units.
+
+        Args:
+            role_arn (String): AWS IAM RoleArn
+        """
+        super().__init__(account)
+        self._auth_cred = self.account.get("authentication")
+        self.client = None
+
+    def _init_session(self):
+        """
+        Set or get a session client for aws organizations
+
+        Args:
+            session = aws session
+        """
+        session = utils.get_assume_role_session(utils.AwsArn(self._auth_cred))
+        session_client = session.client("organizations")
+        LOG.info("Starting aws organizations session for crawler.")
+        self.client = session_client
+
+    def _depaginate(self, function, resource_key, **kwargs):
+        """
+        Depaginates the results of the aws client.
+
+        Args:
+            function (AwsFunction): Amazon function name
+            resource_key (DataKey): Key to grab from results
+            kwargs: Parameters to pass to amazon function
+
+        See: https://gist.github.com/lukeplausin/a3670cd8f115a783a822aa0094015781
+        """
+        response = function(**kwargs)
+        results = response[resource_key]
+        while response.get("NextToken", None) is not None:
+            response = function(NextToken=response.get("NextToken"), **kwargs)
+            results = results + response[resource_key]
+        return results
+
+    def _crawl_accounts_per_id(self, ou, prefix):
+        """
+        List accounts for parents given an aws identifer.
+
+        Args:
+            parent_id: unique id for org unit you want to list.
+            prefix: The org unit tree path
+
+        See:
+        [1] https://docs.aws.amazon.com/cli/latest/reference/organizations/list-accounts-for-parent.html
+        [2] https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/organizations.html
+        """
+        parent_id = ou.get("Id")
+        LOG.info("Obtaining accounts for organizational unit: %s" % parent_id)
+        child_accounts = self._depaginate(
+            function=self.client.list_accounts_for_parent, resource_key="Accounts", ParentId=parent_id
+        )
+        for act_info in child_accounts:
+            self._save_aws_org_method(ou.get("Name", ou.get("Id")), ou.get("Id"), prefix, act_info.get("Id"))
+
+    def _crawl_org_for_acts(self, ou, prefix):
```

Comment:
This is silly, but when I can spare to I try to spell out the full word for readability later on. If I see accounts, I know later what I meant. If I see acts, I actually think of the _Acts of the Apostles_, a book in the Bible, so I have to go through another step to figure out it means accounts.

Reply:
Nah, thats a really good point. Readability is super important, I think it makes more sense to change it to accounts. 👍